### ISSUE TITLE
With cordova-android@9.0.0 defining versionNameSuffix in build-extras.gradle results in suffix being used as complete version name

### ISSUE THREAD

**Issue by cdmahoney** [apache/cordova-android#1025]
# Bug Report

## Problem
Defining versionNameSuffix in build-extras.gradle causes versionNameSuffix to be used as complete versionName with android@9.0.0. This worked correctly with android@8.0.0 

### What is expected to happen?
versionNameSuffix should be appended to config.xml defined version, eg
version="1.0.0"
versionNameSuffix = ".debug"

Compiled application (debug build type) should have version name "1.0.0.debug". 

### What does actually happen?
Actual version name is just versionNameSuffix , eg ".debug"

## Information

To reproduce problem:
- Create new cordova project
- Add platform android@9.0.0
- Add build-extras.gradle shown below.
- Compile and test debug build type.

### Command or Code

Sample build-extras.gradle
```
android {
	buildTypes {
		debug {
			versionNameSuffix ".debug"
		}
	}
}
```

### Environment, Platform, Device

Compiling on Windows 10. 

### Version information

Cordova CLI: 9.0.0 (cordova-lib@9.0.1)
Cordova android 9.0.0

## Checklist

- [x ] I searched for existing GitHub issues
- [x ] I updated all Cordova tooling to most recent version
- [ x] I included all the necessary information above

**Comment by breautek** [comment]
Thank you for taking the time to fill out the issue form. I've Confirmed the issue.

On 8.1.0 I observed:

```
aapt dump badging app-debug.apk 
package: name='io.cordova.hellocordova' versionCode='10000' versionName='1.0.0.debug' platformBuildVersionName='1.0.0.debug'
```

On 9.0.0 I observed:
```
aapt dump badging app-debug.apk 
package: name='io.cordova.hellocordova' versionCode='10000' versionName='.debug' compileSdkVersion='29' compileSdkVersionCodename='10'
```

Not sure on the cause, I don't have a workaround at this time.

**Comment by cdmahoney** [comment]
OK thanks, no problem - for now I've just defined the suffix to be the complete version name in build-extras.gradle. Good enough for the debug version!

Colin

**Comment by breautek** [comment]
This appears to be a regression bug in the `Android Gradle Plugin` version 4.0.0. 

https://issuetracker.google.com/issues/158454676

It's assigned and open, and claims to be effort to get this fixed in `4.0.1`. I'll leave this ticket open as a reminder, so that we can upgrade the framework to the patched version when it becomes available.

**Comment by breautek** [comment]
I've retested this against current stable release and the issue is no longer reproducible. Closing.
